queries/wiki.py

```diff
@@ -0,0 +1,156 @@
+"""
+
+    Reynir: Natural language processing for Icelandic
+
+    Arithmetic query response module
+
+    Copyright (C) 2019 Miðeind ehf.
+
+       This program is free software: you can redistribute it and/or modify
+       it under the terms of the GNU General Public License as published by
+       the Free Software Foundation, either version 3 of the License, or
+       (at your option) any later version.
+       This program is distributed in the hope that it will be useful,
+       but WITHOUT ANY WARRANTY; without even the implied warranty of
+       MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+       GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see http://www.gnu.org/licenses/.
+
+
+    This modules answers queries by fetching information on diverse topics
+    from the Icelandic Wikipedia API.
+
+"""
+
+from . import query_json_api
+from datetime import datetime, timedelta
+import re
+
+_WIKI_QTYPE = "Wikipedia"
+
+# For end user presentation
+_WIKIPEDIA_CANONICAL = "Wikipedía"
+_WIKIPEDIA_VOICE = "Vikipedía"
+
+_WIKI_VARIATIONS = (
+    "vikipedija",
+    "víkípedija",
+    "vikípedija",
+    "víkipedija",
+    "víkípedía",
+    "víkipedía",
+    "vikípedía",
+    "vikipedía",
+    "wikipedia",
+    "wikipedía",
+)
+
+# Indicate that this module wants to handle parse trees for queries,
+# as opposed to simple literal text strings
+HANDLE_TREE = True
+
+# The context-free grammar for the queries recognized by this plug-in module
+GRAMMAR = """
+
+Query →
+    QWikiQuery '?'?
+
+QWikiQuery →
+    "hvað" "segir" QWikipedia "um" QWikiSubject
+    | "hvað" "getur" QWikipedia "sagt" "mér"? "um" QWikiSubject
+    | "hvaða" "upplýsingar" "er" QWikipedia "með" "um" QWikiSubject
+    | "hvaða" "upplýsingum" "býr" QWikipedia "yfir" "varðandi" QWikiSubject
+    | "hvað" "myndi" QWikipedia "segja" "mér"? "um" QWikiSubject
+    | "fræddu" "mig" "um" QWikiSubject
+    # | "flettu" "upp" QWikiSubject "í" QWikipedia
+    # | "hvað" "er" QWikiSubject "samkvæmt" QWikipedia
+    # | "hver" "er QWikiSubject "samkvæmt" QWikipedia
+
+QWikiSubject →
+    Nl_þf
+
+QWikipedia →
+    {0}
+
+$score(+35) QWikiQuery
+
+""".format(
+    " | ".join('"' + v + '"' for v in _WIKI_VARIATIONS)
+)
+
+
+def QWikiQuery(node, params, result):
+    # Set the query type
+    result.qtype = _WIKI_QTYPE
+    result.qkey = result["subject_nom"]
+
+
+def Nl(node, params, result):
```

Comment:
Might be better do define the function `QWikiSubject()` to handle this, since it may become more complicated going forward